### DISCUSSION TITLE
Prevent Gatsby from URL rewriting when reverse proxying

### DISCUSSION THREAD

**Discussion by timroes** [gatsbyjs/gatsby#30965] [Help]
I would like to achieve building a Gatsby page where a reverse proxy (more specifically Netlify in this case) reverse proxy a specific part to a page and Gatsby does not rewrite the URL in the browser when loading that page.

Example: I have a Gatsby page, that has via `src/pages/` the two pages `/landing-nl` and `/landing-us`. Now I have the reverse proxy decide based upon visitor location whether visiting `/landing` should actually serve `/landing-nl` or `/landing-us`. The reverse proxy also works fine, and visiting `/landing` will return the approprite page I want it to. The problem is that as soon as this page is loaded in the browser Gatsby now rewrites the URL from `/landing` to `/landing-nl`. I would like to prevent that behavior and have all those `/landing-x` pages in the browser not rewrite to their actual path but keep the `/landing` URL that the reverse proxy served.

Is there a way to configure this somehow?

**Comment by lintonye** [comment]
This is exactly what I'm trying achieve too. I tried `gatsby-plugin-netlify` which seems to support it but doesn't work:

```js
exports.createPages = async ({ actions }) => {
  const { createPage, createRedirect } = actions
  createRedirect({
    fromPath: "/fr_CA/abc",
    toPath: "/region1/fr/abc",
    statusCode: 200,
  })
}
```

Digging further, I've found that Gatsby injects the path as per [documentation](https://www.gatsbyjs.com/docs/html-generation/#6-inject-page-info-to-cdata). Something like:

```html
<script id="gatsby-script-loader">
/*
<![
  CDATA[ */
    window.pagePath="/region1/fr/abc/";
    window.___webpackCompilationHash="2899d283c7271686bd39";
  /* ]
]>
*/
</script>
```
